test/test-mc-tokens.c

```diff
@@ -259,31 +259,40 @@ static void _test_mc_tokens_error(_mongocrypt_tester_t *tester) {
 }
 
 static void _test_mc_tokens_raw_buffer(_mongocrypt_tester_t *tester) {
-    mc_ServerDataEncryptionLevel1Token_t *token;
+    mc_ServerDataEncryptionLevel1Token_t *token1;
+    mc_ServerDataEncryptionLevel1Token_t *token2;
     _mongocrypt_buffer_t test_input;
+    _mongocrypt_buffer_t original_test_input;
     _mongocrypt_buffer_t expected;
 
     _mongocrypt_buffer_copy_from_hex(&test_input, "6c6a349956c19f9c5e638e612011a71fbb71921edb540310c17cd0208b7f548b");
 
     /* Make a token from a raw buffer */
-    token = mc_ServerDataEncryptionLevel1Token_new_from_buffer(&test_input);
+    token1 = mc_ServerDataEncryptionLevel1Token_new_from_buffer(&test_input);
+    token2 = mc_ServerDataEncryptionLevel1Token_new_from_buffer_copy(&test_input);
 
     /* Assert new_from_buffer did not steal ownership. */
     ASSERT(test_input.owned);
     ASSERT(test_input.len == MONGOCRYPT_HMAC_SHA256_LEN);
 
     _mongocrypt_buffer_copy_from_hex(&expected, "6c6a349956c19f9c5e638e612011a71fbb71921edb540310c17cd0208b7f548b");
 
-    ASSERT_CMPBUF(*mc_ServerDataEncryptionLevel1Token_get(token), expected);
+    ASSERT_CMPBUF(*mc_ServerDataEncryptionLevel1Token_get(token1), expected);
+    ASSERT_CMPBUF(*mc_ServerDataEncryptionLevel1Token_get(token2), expected);
 
     /* Assert new_from_buffer references original buffer instead of a copy. */
+    _mongocrypt_buffer_init(&original_test_input);
+    _mongocrypt_buffer_copy_to(&test_input, &original_test_input);
     test_input.data[0] = '0';
     expected.data[0] = '0';
-    ASSERT_CMPBUF(*mc_ServerDataEncryptionLevel1Token_get(token), expected);
+    ASSERT_CMPBUF(*mc_ServerDataEncryptionLevel1Token_get(token1), expected);
+    ASSERT_CMPBUF(*mc_ServerDataEncryptionLevel1Token_get(token2), original_test_input);
```

Comment:
```suggestion
    // Assert new_from_buffer_copy references a new buffer.
    ASSERT_CMPUINT8(mc_ServerDataEncryptionLevel1Token_get(token2)->data[0], !=, expected.data[0]);
```

Suggest removing `original_test_input`. The assert:
```c
ASSERT_CMPBUF(*mc_ServerDataEncryptionLevel1Token_get(token2), original_test_input);
```

Appears redundant with the assertion above:
```c
ASSERT_CMPBUF(*mc_ServerDataEncryptionLevel1Token_get(token2), expected);
```